src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/PreferStreamAsyncMemoryOverloads.cs

```diff
@@ -230,15 +230,15 @@ private void AnalyzeCompilationStart(CompilationStartAnalysisContext context)
                     {
                         rule = PreferStreamReadAsyncMemoryOverloadsRule;
                         ruleMessageMethod = undesiredReadAsyncMethod.Name;
-                        ruleMessagePreferredMethod = preferredReadAsyncMethod.ToDisplayString(SymbolDisplayFormat.CSharpErrorMessageFormat);
+                        ruleMessagePreferredMethod = string.Join(", ", preferredReadAsyncMethod.GetParameters());
                         ruleMessageMemoryType = memoryType.ToDisplayString(SymbolDisplayFormat.CSharpErrorMessageFormat);
```

Comment:
@carlossanlop Do you know if this variable is redundant?

It's used in the following line:

`context.ReportDiagnostic(invocation.CreateDiagnostic(rule, ruleMessageMethod, ruleMessagePreferredMethod, ruleMessageMemoryType));`

As far as I can tell, the `ruleMessageMethod`, `ruleMessagePreferredMethod`, and `ruleMessageMemoryType` are to be injected (using string.Format or whatever similar) into the rule message. However, the rule message only has {0} and {1}, so it seems that `ruleMessageMemoryType` is redundant.

Reply:
No, I think the redundant argument is the variable `ruleMessageMemoryType`, because it is supposed to represent either `Memory` or `ReadOnlyMemory`, but with the message format does not use that. We only use the method name `{0}` and the preferred method name `{1}` that includes the parameters:

> Change the '**ReadAsync**' method call to use the '**ReadAsync(Memory, CancellationToken)**' overload.

Reply:
> No, I think the redundant argument is the variable `ruleMessageMemoryType`

That's exactly what I meant. I've removed that variable.